src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure implements AreaCalculator, Drawable {
+    private String color;
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public int side() {
+        final int minLength = 1;
+        final int maxLength = 10;
+        return new Random().nextInt(minLength, maxLength);
+    }
```

Comment:
You can implement this method in FigureSupplier class and send for every needing object, every invoke this method create new Random, use field for Random

Reply:
not fixed, remove all figure generation logic from here,create a constructor for color instead of a setter